app/models/miq_queue_worker_base/runner.rb

```diff
@@ -62,9 +73,20 @@ def get_message_via_sql
     end
   end
 
+  def get_message_via_miq_messaging
+    @message_queue ||= Queue.new
+
+    @listener_thread   = nil if @listener_thread && !@listener_thread.alive?
+    @listener_thread ||= Thread.new { miq_messaging_listener_thread }
```

Comment:
Why is this in the middle of the get message method?  Seems like it should be elsewhere (perhaps in the do_work loop or similar).

Additionally, this might be cleaner moved to a method like `ensure_listener_thread`